### PR TITLE
tabs with webkit sheet

### PR DESCRIPTION
**Pull request by mko02**
currently, tabs can open up to a external link.
However, I want it to bring up a sheet instead.

Current issue: ed discussion tab is rerouted, but currently it'll just route to the homepage instead

### CODE REVIEW

**Comment by mko02** [comment]
Currently, clicking on the tab will bring up a sheet and and render the web version.
However, I'm not sure how to pass the auth token, doesnt seem to be the access token
Here is **before** logging in with GT SSO

https://github.com/user-attachments/assets/628d24a8-44a5-49c3-9491-7c7ed64ece84

Here is after logging in, the auth token will be saved even switching between tabs

https://github.com/user-attachments/assets/57bc682f-55bc-4136-a462-d5feca9b8a2d

Another issue is that it takes a hot minute to load the sheet with webkit, but after rendering for the first time, its quick.

**Comment by azooz2003-bit** [comment]
Perhaps we can do a pre-lem load when the app launches to decrease the delay.

File: CanvasPlusPlayground/Views/CourseTabsView.swift [hunk 1]
```diff
@@ -6,24 +6,72 @@
 //
 
 import SwiftUI
+import WebKit
+
+struct WebView: UIViewRepresentable {
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/CourseTabsView.swift:11]: We'll need this to work on Mac too:

```swift
#if os(iOS)
typealias PlatformRepresentable = UIViewRepresentable
#else
typealias PlatformRepresentable = NSViewRepresentable
#endif
```
and then modify the code accordingly

File: CanvasPlusPlayground/Views/CourseTabsView.swift [hunk 2]
```diff
@@ -6,24 +6,72 @@
...
+                }
+            }
         }
         .navigationTitle("Tabs")
         .task {
             await tabsManager.fetchTabs()
         }
+        .sheet(isPresented: $showWebView) {
+            
+            VStack {
```
Comment (rahulon12) [inline CanvasPlusPlayground/Views/CourseTabsView.swift:59]: Use a toolbar here.

```swift
NavigationStack {
    Group {
       if let url = selectedURL {
          WebView(url: url)
        } else {
           // ...
       }
       .toolbar {
           // close button
       }
    }
}
```

**Review by rahulon12 — COMMENTED** [review]
Some small changes

**Comment by mko02** [comment]
sheet on the mac:

Screenshot 2024-11-08 at 6 17 30 PM

**Comment by mko02** [comment]
> Perhaps we can do a pre-lem load when the app launches to decrease the delay.

I will create an issue for this

**Review by rahulon12 — APPROVED** [review]

**Pull request merged**